openghg/standardise/footprint/_paris.py

```diff
@@ -0,0 +1,175 @@
+
+    dim_reorder = ("time", "height", "lat", "lon")
+
+    # Ignore type - dv_rename type should be fine as a dict but mypy unhappy.
+    fp_data = fp_data.rename(**dv_rename)  # type: ignore
+    fp_data = fp_data.rename(**dim_rename)
+
+    fp_data = fp_data.transpose(*dim_reorder, ...)
+
+    for attr, new_attr in attribute_rename.items():
+        if attr in fp_data:
+            fp_data.attrs[new_attr] = fp_data.attrs.pop(attr)
+
+    # Need to read the metadata from the footprints and then store it
+    # Do we need to chunk the footprints / will a Datasource store it correctly?
+    metadata: Dict[str, Union[str, float, List[float]]] = {}
+
+    metadata["data_type"] = "footprints"
+    metadata["site"] = site
+    metadata["domain"] = domain
+    metadata["model"] = model
+
+    # Include both inlet and height keywords for backwards compatability
+    metadata["inlet"] = inlet
+    metadata["height"] = inlet
+    metadata["species"] = species
+
+    if met_model is not None:
+        metadata["met_model"] = met_model
+
+    if network is not None:
+        metadata["network"] = network
+
+    # Check if time has 0-dimensions and, if so, expand this so time is 1D
+    if "time" in fp_data.coords:
+        fp_data = update_zero_dim(fp_data, dim="time")
+
+    fp_time = fp_data["time"]
+
+    # If filepath is a single file, the naming scheme of this file can be used
+    # as one factor to try and determine the period.
+    # If multiple files, this input isn't needed.
+    if isinstance(filepath, (str, Path)):
+        input_filepath = filepath
+    else:
+        input_filepath = None
+
+    start_date, end_date, period_str = infer_date_range(
+        fp_time, filepath=input_filepath, period=period, continuous=continuous
+    )
+
+    metadata["start_date"] = str(start_date)
+    metadata["end_date"] = str(end_date)
+    metadata["time_period"] = period_str
+
+    metadata["max_longitude"] = round(float(fp_data["lon"].max()), 5)
+    metadata["min_longitude"] = round(float(fp_data["lon"].min()), 5)
+    metadata["max_latitude"] = round(float(fp_data["lat"].max()), 5)
+    metadata["min_latitude"] = round(float(fp_data["lat"].min()), 5)
+
+    if high_spatial_resolution:
+        try:
+            metadata["max_longitude_high"] = round(float(fp_data["lon_high"].max()), 5)
+            metadata["min_longitude_high"] = round(float(fp_data["lon_high"].min()), 5)
+            metadata["max_latitude_high"] = round(float(fp_data["lat_high"].max()), 5)
+            metadata["min_latitude_high"] = round(float(fp_data["lat_high"].min()), 5)
+
+        except KeyError:
+            raise KeyError("Expected high spatial resolution. Unable to find lat_high or lon_high data.")
+
+    metadata["time_resolved"] = str(time_resolved)
+    metadata["high_spatial_resolution"] = str(high_spatial_resolution)
+    metadata["short_lifetime"] = str(short_lifetime)
+
+    metadata["heights"] = [float(h) for h in fp_data.height.values]
```

Comment:
This metadata ("heights", "variables") comes from the data itself, so should be set in the parser, but "time_resolved", "high_spatial_resolution", "short_lifetime" don't need to be here. 

Also there is a lot of repeated code between this and acrg parser (much of it todo with metadata). Maybe open an issue about refactoring the footprint parsers. I suspect removing unnecessary metadata parsing will remove some of this repeated code. For other parts, it should be moved to helper functions to be shared between parsers.

Reply:
For this PR, the code in the new parser is very close to the code in the existing parser, so it's just another place where probably code will need to be maintained in parallel. Since we're planning to change the parsers soon, we don't have to cover in this PR. (I realise you didn't write the bulk of this code, but we should try to avoid copying and pasting large sections of code so that we don't get into the `Footprint`, `Flux`, `Surface`, etc... issue again.)

Here's an issue for it: https://github.com/openghg/openghg/issues/1082

Reply:
Yes I understood what you were saying, it's more that I don't think we'd want to spend more time on this PR now given we were trying to just get this code in and not let it sit on a branch people were already using. I agree on avoiding repeated code in general but at this stage I was saying it was better to get this in with an improvement on a separate PR if needed - at least this can be tracked on an issue now if needed. I'm going to resolve this.